sys/userlib/src/lib.rs

```diff
@@ -319,6 +334,25 @@ pub fn sys_recv(
     }
 }
 
+/// Convenience wrapper for `sys_recv` for the specific, but common, task of
+/// listening for notifications. In this specific use, it has the advantage of
+/// never panicking and not returning a `Result` that must be checked.
+#[inline(always)]
+pub fn sys_recv_notification(notification_mask: u32) -> u32 {
+    match sys_recv(&mut [], notification_mask, Some(TaskId::KERNEL)) {
+        Ok(rm) => {
+            // The notification bits come back from the kernel in the operation
+            // code field.
+            rm.operation
+        }
+        Err(_) => {
+            // Because we passed Some(TaskId::KERNEL), this is defined as not
+            // being able to happen.
+            unsafe { core::hint::unreachable_unchecked() }
```

Comment:
tiny nit: should this be 
```suggestion
            // Safety: Because we passed Some(TaskId::KERNEL), this is defined
            // as notbeing able to happen.
            unsafe { core::hint::unreachable_unchecked() }
```